app/src/main/java/com/example/una/ScrollListener/EndlessRecyclerViewScrollListener.java

```diff
@@ -0,0 +1,106 @@
+package com.example.una.ScrollListener;
+
+import androidx.recyclerview.widget.GridLayoutManager;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.recyclerview.widget.StaggeredGridLayoutManager;
+
+public abstract class EndlessRecyclerViewScrollListener extends RecyclerView.OnScrollListener {
+    // The minimum amount of items to have below your current scroll position
+    // before loading more.
+    private int visibleThreshold = 5;
+    // The current offset index of data you have loaded
+    private int currentPage = 0;
```

Comment:
your scroll listener starts with page 0. wouldn't that cause page 1 to load twice?